pro-planner-client/src/components/Poll.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import Options from "./Options";
+import {Button, Card, Col, Container, Row} from "react-bootstrap";
+import {useState} from 'react';
+import AddOptionForm from "./AddOptionForm";
+import {BiCaretDown, BiCaretUp} from "react-icons/bi";
+import {useDispatch} from 'react-redux';
+import {voteOption} from "../redux/pollSlice";
+
+
+function Poll({poll}) {
+
+    const [showModal, setShowModal] = useState(false);
+    const [isOptionListDisplay, setIsOptionListDisplay] = useState(false);
+    const [selectedOptions, setSelectedOptions] = useState([]);
+    const dispatch = useDispatch();
+
+    const caretIconSize = 25;
+
+    const handleAddOption = () => {
+        setShowModal(true);
+    };
+
+    const handleIconClick = () => {
+        if (!isOptionListDisplay) {
+            setIsOptionListDisplay(true);
+        } else {
+            setIsOptionListDisplay(false);
+        }
+    }
+
+    // const userId = "user"; TODO... userID from backend
+
+    let formResult = {
+        pollId: poll.pollId,
+        selectedOptions: selectedOptions,
+        // userId: userId TODO...
+    }
+
+    const handleVote = () => {
+        // TODO... implement error message if user has already voted
```

Comment:
Noting the redux state, I think users can vote on a poll more than once. For example, if they wish to change their vote they can vote again. I would suggest either they can see the options they have voted for pre-checked. Or voting again would overwrite their existing vote. Note that for this to be possible each option needs to track who has voted for it.

Reply:
Noted! Just to confirm, users can vote for only for one option per poll right?

Reply:
I think for simplicity, we can just go with one for now. If you wish to you can also work on multiple votes, but that's extra.